### PR TITLE
release v2

### PR DESCRIPTION
**Pull request by MulyukovAidar**

### CODE REVIEW

File: src/settings.ts [hunk 1]
```diff
@@ -24,33 +24,66 @@
...
+
+
+class TextBoxSettingsCard extends Card {
+
+    name: string = "textBox";
+    displayNameKey?: string = "Visual_Textbox_Settings";
+    placeholderTextKey: string = "Visual_Input_Placeholder"
+
+
+    private minFontSize: number = 8;
```
Comment (uve) [inline src/settings.ts:51]: Is it a constant variable?

File: src/settings.ts [hunk 2]
```diff
@@ -24,33 +24,66 @@
...
+    private defaultFontSize: number = 11;
+
+    enableBorder = new ToggleSwitch({
```
Comment (uve) [inline src/settings.ts:54]: set public or private type to make it explicit

File: src/visual.ts [hunk 3]
```diff
@@ -25,51 +25,58 @@
...
-  private searchButton: d3.Selection<HTMLButtonElement, any, any, any>;
-  private clearButton: d3.Selection<HTMLButtonElement, any, any, any>;
+  private searchUi: d3Selection<HTMLDivElement, any, any, any>;
+  private searchBox: d3Selection<HTMLInputElement, any, any, any>;
+  private searchButton: d3Selection<HTMLButtonElement, any, any, any>;
+  private clearButton: d3Selection<HTMLButtonElement, any, any, any>;
   private column: powerbi.DataViewMetadataColumn;
   private host: powerbi.extensibility.visual.IVisualHost;
-  private settings: VisualSettings;
+  // private settings: VisualSettings;
```
Comment (uve) [inline src/visual.ts:60]: remove this comment

File: src/visual.ts [hunk 4]
```diff
@@ -86,39 +93,57 @@ export class Visual implements IVisual {
...
 
+  public getFormattingModel(): powerbi.visuals.FormattingModel {
+    // removes border color
+    if (this.formattingSettings?.textBox.enableBorder.value === false) {
+      this.formattingSettings.removeBorderColor();
+    }
+    const model = this.formattingSettingsService.buildFormattingModel(this.formattingSettings);
+
+    // hotfix for composite slice
+    const buggedFontControl = (<any>model.cards[0]).groups[0].slices.find(slice => !slice.displayName && slice.control.type === "FontControl")
```
Comment (uve) [inline src/visual.ts:132]: replace it with  (<any>model.cards[0])?.groups[0]?.slices?.

File: pbiviz.json [hunk 5]
```diff
@@ -4,13 +4,13 @@
...
     "description": "Provides a search box that can be placed anywhere in your dashboard. This adds a text filtering capability for quick searching across your data.",
     "supportUrl": "https://community.powerbi.com",
     "gitHubUrl": "https://github.com/microsoft/PowerBI-visuals-TextFilter"
   },
   "output": "dist/textFilter.pbiviz",
-  "apiVersion": "3.4.0",
+  "apiVersion": "5.3.0",
   "author": {
     "name": "Microsoft",
     "email": "msrvizsupport@microsoft.com"
```
Comment (uve) [inline pbiviz.json:16]: replace e-mail with pbicvsupport

**Review by uve — CHANGES_REQUESTED** [review]

File: src/settings.ts (hunk 1 above)
Comment (MulyukovAidar) [reply src/settings.ts:51]: yes

File: src/visual.ts (hunk 3 above)
Comment (MulyukovAidar) [reply src/visual.ts:60]: done

**Pull request closed without merging**